### PR TITLE
Disable colors for non-interactive output

### LINKED ISSUE #480: Default color output should depend on TTY
# This is a Feature Proposal

## Description

When piping the default webpack output into a file (or when using CI/CD tool without shell color escape support), color escapes are added. The plugin should first check if the output is a tty before writing colors (this is a more sensible default IMO). It can still be overridden explicitly, but by default the plugin should decide if the output supports colors before writing them.

## Additional Data

* ***Serverless-Webpack Version you're using***: 5.2.0
* ***Webpack version you're using***: 4.16.4
* ***Serverless Framework Version you're using***: 1.30.0
* ***Operating System***: OSX

### PR DESCRIPTION
**Pull request by c0d3d**
## What did you implement:

Closes #480 

## How did you implement it:

Check if stdout is a tty when deciding to use colors or not.

## How can we verify it:

1. Run from tty (interactive shell) to see default still has colors
2. Pipe into a file to see that it doesn't write colors.

## Todos:

- [X] Write tests
- [X] Write documentation
- [X] Fix linting errors
- [X] Make sure code coverage hasn't dropped
- [X] Provide verification config / commands / resources
- [X] Enable "Allow edits from maintainers" for this PR
- [X] Update the messages below

***Is this ready for review?:*** YES
***Is it a breaking change?:*** NO

### CODE REVIEW

**Comment by c0d3d** [comment]
@HyperBrain Can you look at this? Would help our Jenkins build from looking awful.
Thanks!

**Comment by c0d3d** [comment]
@hassankhan Can you help get this merged?
Would make my Jenkins output much better :D

**Comment by hassankhan** [comment]
@c0d3d Unfortunately I don't have write access to the repo, I would love to see this PR merged for the same reason 👍

**Comment by HyperBrain** [comment]
@hassankhan I just invited you to the serverless-webpack-team. You should have write access as soon as you confirm the request ;-)

File: lib/compile.js [hunk 1]
```diff
@@ -3,6 +3,7 @@
 const _ = require('lodash');
 const BbPromise = require('bluebird');
 const webpack = require('webpack');
+const tty = require('tty');
```
Comment (HyperBrain) [inline lib/compile.js]: The `tty` package is not part of the project's devDependencies. Can you add it to the package.json?

Comment (c0d3d) [reply lib/compile.js]: The `tty` package is actually a node built-in. The npm package is a security holder now.

Comment (hassankhan) [reply lib/compile.js]: Resolving since this is fine 👍

**Comment by hassankhan** [comment]
Thanks for the invite @HyperBrain 😄! If there's no other issue with the PR, is it cool to go ahead and merge it? Or is there any other process I should be following aside from code-review + passing tests?

**Comment by HyperBrain** [comment]
No, that should be ok. However, it is important to have the milestone set
properly on the PR and the issue, so that I can create a proper changelog
for the next release

**Comment by hassankhan** [comment]
Thanks for creating the PR @c0d3d, should be released in v5.3.0 👍

**Pull request merged**